source/blog/2021-08-09-opal-1-2.html.md

```diff
@@ -0,0 +1,178 @@
+
+### Issues
+
+Opal 1.2 is mostly a bugfix release. We fixed a lot of long-standing bugs. Maybe it's time to try Opal again? Here we have a chart:
+
+[![Issues chart](https://i.imgur.com/cRm19Tf.png)](https://i.imgur.com/cRm19Tf.png)
+
+## Roadmap
+
+We didn't fully meet the expectations of the roadmap as described in the Opal 1.0 post. Better than postpone the Opal 1.1 release, we decided to release what we have (and we had quite a lot).
+
+We have partially accomplished the code minification with an external (and experimental) [opal-optimizer gem](https://github.com/hmdne/opal-optimizer).
+
+There are solutions for Webpack integration, but those aren't upstream. We want to gladly invite anyone interested to contribute during the Opal 1.3 development phase which we mainly want to dedicate towards accomplishment of that goal.
+
+Opal 2.0 will hide Ruby methods behind a (JavaScript) symbol wall increasing compatibility with JavaScript libraries.
+
+String are still immutable, (Regexps are still somewhat incompatible)[https://opalrb.com/blog/2021/06/26/webassembly-and-advanced-regexp-with-opal/], asynchronous methods can't be implemented in a Ruby synchronous style. Those features we plan for Opal 2.0 or later on, as they will require a lot of changes.
+
+## Ecosystem
+
+All Gems from the opal-* namespace are compatible with the Opal 1.2 release. The new opal-browser version is up for a release soon, until then, we recommend you use the `master` branch!
+
+From the outside of the core ecosystem, we have seen quite a lot of development, there are at least a few very interesting things that were done using Opal:
+
+* [Glimmer DSL for Opal](https://github.com/AndyObtiva/glimmer-dsl-opal) - an (experimental) platform-agnostic (web with Opal, desktop with SWT or Tk) toolkit for graphical applications. Write once, run anywhere!
+* [18xx.games](https://18xx.games) - a website where you can play async or real-time 18xx games
+* [Hyperstack](https://hyperstack.org/) - a well documented and stable Ruby DSL for wrapping React and ReactRouter as well as any JavaScript library or component. No need to learn JavaScript!
+
+## Conclusion
+
+If you are unsure yet, Opal doesn't just _feel_ like Ruby, it _IS_ Ruby. It is absolutely possible to create a website with pure Ruby and not a single line of JavaScript (and a good performance!). In the advent of solutions like Hotwire, Opal goes a step higher and allows you to create fully isomorphic websites (with a fully fledged support for human language alike DSLs). Forget context switches from CoffeeScript to Ruby when you develop a feature, it's all Ruby!
```

Comment:
```suggestion
If you are unsure yet, Opal doesn't just _feel_ like Ruby, it _IS_ Ruby. It is absolutely possible to create a website with pure Ruby and not a single line of JavaScript (and a good performance!). In the advent of solutions like Hotwire, Opal goes a step higher and allows you to create fully isomorphic websites (with a fully fledged support for human language alike DSLs). Forget context switches from JavaScript, TypeScript, or even CoffeeScript to Ruby when you develop a feature, it's all Ruby!
```

CoffeeScript is not very liked nowadays and also less likely to be the compiler of choice for most application, so let's add a couple that you'll likely encounter in a modern Rails app.